woocommerce/sensei-course-progress
# Add build process to aid in deployment

This PR adds a Gulp-based build process that facilitates deployment. It includes the following tasks:

- `gulp clean` - Deletes the `/build` folder.
- `gulp css` - Compiles and minifies `.scss` files to `/build` folder.
- `gulp php` - Copies PHP files to `/build` folder.
- `gulp languages` - Copies files in `languages` folder to `/build` folder.
- `gulp watch`- Watches for changes to any of the CSS, PHP or language files and automatically calls the appropriate task runners when there are changes.
- `gulp`, `gulp build` - Runs `clean`, `css`, `php`, and `languages` tasks to create a `/build` folder that is ready for deployment.

## Testing
- Run `npm install` and `gulp build`. Check that the `/build` folder is created.
- Copy the contents of the `/build` folder to `wp-content/plugins/sensei-course-progress`.
- Browse to the widget on the front-end and ensure that it is styled correctly and that the CSS file is loaded.
